src/Site.js

```diff
@@ -385,6 +412,193 @@ Site.prototype.createPage = function (config) {
   });
 };
 
+/**
+ * Converts an existing Github wiki or docs folder to a MarkBind website.
+ */
+Site.prototype.convert = function (rootPath) {
+  return new Promise((resolve, reject) => {
+    this.addIndexPage(rootPath)
+      .then(() => this.addAboutPage(rootPath))
+      .then(() => this.addFooterToDefaultLayout(rootPath))
+      .then(() => this.addSiteNavToDefaultLayout(rootPath))
+      .then(() => this.addDefaultLayoutToSiteConfig(rootPath))
+      .then(() => this.addTopNavToAddressablePages(rootPath))
+      .then(resolve)
+      .catch(reject);
+  });
+};
+
+/**
+ * Adds an index page to site if not present.
+ * @param rootPath
+ */
+Site.prototype.addIndexPage = function (rootPath) {
+  const indexPagePath = path.join(rootPath, INDEX_MARKDOWN_FILE);
+  const fileNames = ['README.md', 'Home.md'];
+  for (let i = 0; i < fileNames.length; i += 1) {
+    const filePath = path.join(rootPath, fileNames[i]);
+    if (fs.existsSync(filePath)) {
+      fs.copyAsync(filePath, indexPagePath)
+        .catch(() => Promise.reject(new Error(`Failed to copy over ${fileNames[i]}`)));
+      break;
+    }
+  }
+  return Promise.resolve();
```

Comment:
This function resolves before `copyAsync` resolves. Maybe you should declare a variable above and reassign it to the new promise, after which you can return it at the end of the function?

```js

let promise = Promise.resolve();
...
// logic for copying file
promise = fs.copyAsync(filePath, indexPagePath)
        .catch(() => Promise.reject(new Error(`Failed to copy over ${fileNames[i]}`)));
...
return promise;
```

Reply:
@nicholaschuayunzhi Nice catch! It's clearer to `return fs.copyAsync(...).catch(...);` directly.